pkgs/by-name/po/porn-vault/package.nix

```
@@ -0,0 +1,103 @@
+{
+  fetchFromGitLab,
+  fetchurl,
+  rustPlatform,
+  lib,
+  pnpm,
+  stdenvNoCC,
+  nodejs_22,
+  ffmpeg,
+  imagemagick,
+  writeShellScript,
+  makeWrapper,
+  autoPatchelfHook,
+}:
+let
+  izzy = rustPlatform.buildRustPackage rec {
+    pname = "izzy";
+    version = "2.0.1";
+
+    src = fetchFromGitLab {
+      owner = "porn-vault";
+      repo = "izzy";
+      rev = version;
+      hash = "sha256-UauA5mZi5a5QF7d17pKSzvyaWbeSuFjBrXEAxR3wNkk=";
+    };
+
+    cargoHash = "sha256-y0bV83bnuSALwvRD8ztOiBE6LSg3Hx6HKSjvQPEkvIQ=";
+    cargoPatches = [ ./add-Cargo.lock.patch ];
+
+    meta = with lib; {
+      description = "Rust In-Memory K-V Store with Redis-Style File Persistence and Secondary Indices";
+      homepage = "https://gitlab.com/porn-vault/izzy";
+      license = lib.licenses.gpl3;
+      maintainers = with maintainers; [ luNeder ];
+      platforms = lib.platforms.all;
+      mainProgram = "izzy";
+    };
+  };
+in
```

```suggestion
  nodejs = nodejs_22;
in
```

We can also do something similar with nodejs to avoid specifying the version number again and again